synapse/handlers/federation.py

```diff
@@ -197,12 +217,36 @@ async def maybe_backfill(
                 return. This is used as part of the heuristic to decide if we
                 should back paginate.
         """
+        processing_start_time = self.clock.time_msec()
```

Comment:
Starting the processing time here so we can include the room backfill linearizer in the timing.

> The linearizer is only used to avoid repeating work at the same time but is very simplistic. We can optimize how this works in the future so it would be good to have it as part of the before timing.